### PR TITLE
removed from calendar

### PR DESCRIPTION
**Pull request by hn275**
# Remove From Calendar

Implemented the endpoint for removing a course from user's cookie.

Here's the curl command to remove CSC 429 from cookie:

```sh
curl \
    -H "Content-Type: application/x-www-form-urlencoded" \
    -X DELETE "http://localhost:8080/calendar" \
    -d "course_code=429&subject_code=CSC&course=CSC%20429"
```

### CODE REVIEW

**Comment by brennanmcmicking** [comment]
i'm waiting for #26 before reviewing this

**Comment by malcolmseyd** [comment]
Sorry Hal, I'm gonna cause a massive merge conflict

**Comment by hn275** [comment]
@brennanmcmicking @malcolmseyd I'm drafting this for now, until #26 is merged then I'll resolve the diff/make it work with new updated flow.

**Comment by malcolmseyd** [comment]
You are now unblocked :magic_wand: 
(pls rebase on main thank u)

**Comment by hn275** [comment]
> You are now unblocked 🪄 (pls rebase on main thank u)

rebased, authored, ready for review, etc etc

**Comment by malcolmseyd** [comment]
Just curious, how did you trigger this endpoint when testing? I can't find any elements on the page to trigger it so I'm having to fall back to curl

File: src/routes/calendar.rs [hunk 1]
```diff
@@ -56,10 +56,39 @@ pub async fn add_to_calendar<'a, 'b>(
...
+
+        selected_courses
+            .courses
+            .iter()
+            .filter(|&(thin_course, _)| thin_course.course_code != course.course_code)
+            .for_each(|(course, section)| {
+                user_state
+                    .courses
+                    .insert(course.to_owned(), section.to_owned());
+            });
```
Comment (malcolmseyd) [inline src/routes/calendar.rs]:
```suggestion
        let mut user_state = selected_courses;
        user_state
            .courses
            .retain(|thin_course, _| thin_course.course_code != course.course_code);
```

**Review by malcolmseyd — DISMISSED** [review]
Looks good to me! I just have one suggestion to make the code shorter but LGTM otherwise :+1:

**Comment by malcolmseyd** [comment]
After this gets merged, we should get a UI component ASAP to actually use this.

I personally didn't like testing this PR manually with curl, but I also understand that UI was technically out of scope. From here on out, I feel like data-layer things should be written after or at the same time as UI code so we that can test it more easily. Remember, we're not building an HTML API or a "backend", we're building an HTML application, so the whole thing should be QA-able just using the browser and maybe devtools.

**Review by malcolmseyd — APPROVED** [review]
LeLGTM

**Pull request merged**